src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,34 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  onTodoClear: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todo,
+  onTodoClear: todoCleaner,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getUser(todo.userId)
+      .then(setUser)
+      .finally(() => setLoading(false));
+  }, [todo.userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
```

Comment:
According to the checklist, you should avoid interacting with the DOM directly and use React as much as possible. In this case, you are using 'className' directly. You could use the 'classnames' library to define classes conditionally. It's a cleaner and more readable way to handle conditional classes in React.